test/parallel/test-http-should-keep-alive.js

```diff
@@ -41,34 +43,32 @@ const SHOULD_KEEP_ALIVE = [
   true,  // HTTP/1.1, Connection: keep-alive
   false  // HTTP/1.1, Connection: close
 ];
-let requests = 0;
-let responses = 0;
 http.globalAgent.maxSockets = 5;
 
+const countdown = new Countdown(SHOULD_KEEP_ALIVE.length , () => server.close());
+let countdownIndexInc = SERVER_RESPONSES.length - countdown.remaining;
+
 const server = net.createServer(function(socket) {
-  socket.write(SERVER_RESPONSES[requests]);
-  ++requests;
+  socket.write(SERVER_RESPONSES[countdownIndexInc]);
 }).listen(0, function() {
   function makeRequest() {
     const req = http.get({ port: server.address().port }, function(res) {
       assert.strictEqual(
-        req.shouldKeepAlive, SHOULD_KEEP_ALIVE[responses],
-        `${SERVER_RESPONSES[responses]} should ${
-          SHOULD_KEEP_ALIVE[responses] ? '' : 'not '}Keep-Alive`);
-      ++responses;
-      if (responses < SHOULD_KEEP_ALIVE.length) {
+        req.shouldKeepAlive, SHOULD_KEEP_ALIVE[countdownIndexInc],
+        `${SERVER_RESPONSES[countdownIndexInc]} should ${
+          SHOULD_KEEP_ALIVE[countdownIndexInc] ? '' : 'not '}Keep-Alive`);
+      countdown.dec();
+      if (countdown.remaining) {
         makeRequest();
-      } else {
-        server.close();
       }
       res.resume();
     });
   }
-
   makeRequest();
 });
 
 process.on('exit', function() {
-  assert.strictEqual(requests, SERVER_RESPONSES.length);
-  assert.strictEqual(responses, SHOULD_KEEP_ALIVE.length);
+  countdownIndexInc = SERVER_RESPONSES.length - countdown.remaining;
+  assert.strictEqual(countdownIndexInc, SERVER_RESPONSES.length);
+  assert.strictEqual(countdownIndexInc, SHOULD_KEEP_ALIVE.length);
```

Comment:
This block can be removed altogether. The process won't exit unless the countdown finishes.